glusterd2/commands/volumes/volume-profile.go

```diff
@@ -0,0 +1,312 @@
+package volumecommands
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/servers/sunrpc/dict"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+
+	"github.com/gorilla/mux"
+)
+
+func registerVolProfileStepFuncs() {
+	transaction.RegisterStepFunc(txnVolumeProfile, "volume.Profile")
+}
+
+func volumeProfileHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	volname := mux.Vars(r)["volname"]
+	option := mux.Vars(r)["option"]
+
+	txn, err := transaction.NewTxnWithLocks(ctx, volname)
+	if err != nil {
+		status, err := restutils.ErrToStatusCode(err)
+		restutils.SendHTTPError(ctx, w, status, err)
+		return
+	}
+	defer txn.Done()
+
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
```

Comment:
StatusInternalServerError might not be the right status, since the volume name passed may not be the right volume name, it may not exists.

Reply:
I am still not sure what would be appropriate here. Any suggestions ?

Reply:
as @rishubhjain pointed we need to change this also you can use ```restutils.ErrToStatusCode(err)``` to get status code from  error message.